File: typescript/base-hmi-types/src/screens/shapes/HmiCircleSegment.ts
import { HmiProperty, staticProperty } from "../base.js";
import { HmiCircle } from "./HmiCircle.js";
import { HmiObjectType } from "../base/HmiObjectType.js";

export class HmiCircleSegment extends HmiCircle {
  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiCircleSegment;
  }

  startAngle: HmiProperty<number> = staticProperty(0);
  sweepAngle: HmiProperty<number> = staticProperty(0);
}

File: src/BaseHmiTypes/Screens/Shapes/HmiCircleSegment.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Shapes;

public class HmiCircleSegment : HmiCircle
{
    public HmiCircleSegment()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiCircleSegment;
    }

    public HmiProperty<double> StartAngle { get; set; } = 0;

    public HmiProperty<double> SweepAngle { get; set; } = 0;
}
